Less use of legacy mortar mechanical contact constraints

recuero opened the pull request:
- Remove and migrate tests exercising legacy contact (leaving some for coverage and use guidance)
- Remove legacy contact from action

Closes #20773.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1012231/) on 33d5891 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20776/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1012239/) on 33d5891 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>3587c9</th>
      <th colspan=4>#20776 33d589</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.56%</td>
      <td>82.56%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>74149</td>
      <td>74150</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15661</td>
      <td>15660</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20776/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20776/coverage/framework)


## Modules coverage

### Contact

<table>
  <thead>
    <tr>
      <th></th>
      <th>3587c9</th>
      <th colspan=4>#20776 33d589</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.01%</td>
      <td>79.25%</td>
      <td>-9.77%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>2917</td>
      <td>2597</td>
      <td>-320</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>360</td>
      <td>680</td>
      <td>+320</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20776/coverage_diff/contact.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20776/coverage/contact)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20776/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1012239/) on 33d5891 wanted to post the following:


The following coverage requirement(s) failed:
- `contact` coverage rate 79.25% is less than the required 87.0%

recuero committed:
Migrate and remove some legacy mortar contact tests (#20773)

recuero committed:
Remove legacy mortar contact option from contact action (#20773)

recuero committed:
Add proper ad global indexing to migrated tests (#20773)

recuero committed:
Remove dead code from the action.

We are not allowing monomials in the action anymore when doing mortar contact due to the nature of the constraints.

lindsayad reviewed (approved):
Looks clean to me
